src/middlewares/auth.middleware.ts

```diff
@@ -0,0 +1,56 @@
+import {Request, Response, NextFunction} from "express";
+import {JWT} from "@/config";
+import jwt from "jsonwebtoken";
+
+declare module "express-serve-static-core" {
+    interface Request {
+        user?: {id: number, role: string, email?: string}
+    }
+}
+
+export async function authMiddleware(
+    req: Request, 
+    res:Response, 
+    next: NextFunction){
+
+        try{
+            const authHeader = req.headers["authorization"];
+
+            if (!authHeader || !authHeader.startsWith("Bearer ")){
+                return res.status(401).json({message: "Authorization token required"});
+            }
+
+            const token = authHeader.split( " " ) [1];
+
+            const decodedToken = jwt.verify(token, JWT.secret) as {
+                id: number;
+                role: string;
+                email?: string;
+            };
+
+            req.user = {id: decodedToken.id, role: decodedToken.role, email: decodedToken.email};
+
+            next();
+
+        } catch (error) {
+            console.error("Auth.error:", error);
+            
+            return res.status(401).json({message: "Invalid of expired token"});
```

Comment:
invalid or expired token.

make this more robust. dont lump the two errors. you can check the instance of the error matches token expired error (check it up), so you return a response for user to sign in again..then for any other error you can give back invalid token or so